src/app/api/crawl/seed.ts

```diff
@@ -36,13 +36,19 @@ async function seed(url: string, limit: number, indexName: string, options: Seed
     const documents = await Promise.all(pages.map(page => prepareDocument(page, splitter)));
 
     // Create Pinecone index if it does not exist
-    const indexList = await pinecone.listIndexes();
-    const indexExists = indexList.some(index => index.name === indexName)
+    const indexList: string[] = (await pinecone.listIndexes())?.indexes?.map(index => index.name) || [];
+    const indexExists = indexList.includes(indexName);
     if (!indexExists) {
       await pinecone.createIndex({
         name: indexName,
         dimension: 1536,
         waitUntilReady: true,
+        spec: { 
+          serverless: { 
+              cloud: 'aws', 
```

Comment:
The values on this line and the one after it shouldn't be hard coded, but rather defined as environment variables.

Reply:
@rschwabco Thanks! Made the change and tested it.